### ISSUE TITLE
Support TS custom switch-case indentation

### ISSUE THREAD

**Issue by mjbvz** [microsoft/vscode#180239]
Support issue for #179670

See https://github.com/microsoft/TypeScript/pull/53733 for details

**Comment by mjbvz** [comment]
Fixed by  #179670

**Comment by mjbvz** [comment]
To verify:

1.  `npm install typescript@next` in the workspace
2. Switch the workspace to use that version
3. Create a simple switch case statement
4. Confirm that by default, `case` clauses have an extra indent when formatted:

```ts
switch ('') {
    case '': break;
}
```

5. Set `"typescript.format.indentSwitchCase": false`
6. Now confirm that formatting changes the code to remove the extra indent:

```ts
switch ('') {
case '': break;
}
```
